JerryJeager/Will-Be-There
# Resolves #54 - Migrating landing page files from JavaScript to TypeScript

# Goal
Resolves #54 
This pull request resolves issue #54. The goal was to migrate all files related to the landing page from JavaScript to TypeScript. All changes have been made, except for the `NumberMetrics.jsx` file, as the `ScrollTrigger` was not working correctly after migrating to TypeScript.

# Testing

To test this pull request:

1. Run the local server with the new changes (`npm run dev`).
2. Check all parts of the landing page to ensure everything is functioning correctly.
3. Pay attention to the browser console for any TypeScript-related errors.
4. Ensure that `NumberMetrics.jsx` still works as expected in the JavaScript version.

# Checklist

- [x] This PR is connected to issue #54
- [x] PR name follows the pattern **Resolves [Ticket Name]**
- [x] All relevant files on the landing page have been migrated to TypeScript
- [x] Testing has been successfully completed without errors
- [x] `NumberMetrics.jsx` has remained in JavaScript due to issues with `ScrollTrigger` in TypeScript
